src/Sushi.MicroORM/Mapping/DataMapProvider.cs

```diff
@@ -35,8 +35,8 @@ public class DataMapProvider
         public void AddMapping(Type classToMap, Type dataMap)
         {
             //check if class to map has a default, parameterless constructor
-            if (classToMap.GetConstructor(Type.EmptyTypes) == null)
-                throw new ArgumentException($"{classToMap} does not have a parameterless constructor", nameof(classToMap));
+           // if (classToMap.GetConstructor(Type.EmptyTypes) == null)
```

Comment:
I don't think this check should be removed, but it should work with private constructors as well